spec/formatting.md

```diff
@@ -1,24 +1,113 @@
-# WIP DRAFT MessageFormat 2.0 Formatting Behaviour
+# DRAFT MessageFormat 2.0 Formatting Behaviour
 
 ## Introduction
 
 This document defines the behaviour of a MessageFormat 2.0 implementation
 when formatting a message for display in a user interface, or for some later processing.
 
-The document is part of the MessageFormat 2.0 specification,
-the successor to ICU MessageFormat, henceforth called ICU MessageFormat 1.0.
+Message formatting is defined in terms of the following stages:
 
-## Literal Resolution
+- **_Resolution_** determines the value of a part of the message,
```

Comment:
You might be missing an "assignment" stage (processing of `let` statements), which occurs before selection (and might contain some "resolutions")?

Reply:
That's done implicitly through variable resolution. There's also this on line 736:

> Resolution and Formatting errors in _expressions_ that are not used
> in _pattern selection_ or _formatting_ MAY be ignored
> as such do not impact the current message's formatting.

In other words, an "assignment" stage is explicitly left out to allow for implementations that either:
1. Eagerly resolve all declarations immediately, or
2. Lazily only resolve declarations that are required by the message.

If we did include an assignment stage, we would effectively mandate the former even when an implementation could determine that it never needed the value of a declaration.